packages/manager/modules/sign-up/src/form/details/details.controller.js

```diff
@@ -78,12 +78,14 @@ export default class SignUpDetailsCtrl {
     SignUpDetailsCtrl.setElementFocus(fieldName);
   }
 
-  preselectLanguage() {
-    if (get(this.signUpFormCtrl, 'rules.language.in.length', 0) === 1) {
-      const [uniqueLanguage] = this.signUpFormCtrl.rules.language.in;
-      this.signUpFormCtrl.model.language = uniqueLanguage.value;
-    } else {
-      this.signUpFormCtrl.model.language = null;
+  preselectLanguage(isOnChange) {
+    if (isOnChange) {
+      if (get(this.signUpFormCtrl, 'rules.language.in.length', 0) === 1) {
```

Comment:
Can you keep this if block ouside of the `if (isOnChange)` as it allow us to preselect the language (if unique) on the arrival on the page